corpdir-srv-jpa/src/main/java/com/github/bordertech/corpdir/jpa/util/MapperUtil.java

```diff
@@ -178,8 +178,12 @@ public static String cleanApiKey(final String key) {
 		if (id == null) {
 			return null;
 		}
-		Long longId = convertApiIdforEntity(id);
-		return getEntityById(em, longId, clazz);
```

Comment:
I think the logic being added here is adding to the confusion of the `getEntityByBusinessKey` return type.

The logic of `isEntityId` is catered for in `getEntityByKeyId` which is called when you are not sure if you have `id` or `businessKey`.  But `getEntityByApiId` assumes the ID being passed in is already an entity id.

`getEntityByBusinessKey` assumes it is being passed a business key and will return an object with a business key.

Also, pretty sure `UnitType` has a `businessKey`.

Reply:
With the `codacy` issue, what if you try moving the field defintions above the `constrcutor`. I wonder if `codacy` modified the PMD rule.

Reply:
Yes you are correct, `AbstractJpaIdService.getEntity`, calls `MapperUtil.getEntityByApiId` which always expects argument to be `id`, whereas `AbstractJpaKeyIdService.getEntity` calls `MapperUtil.getEntityByKeyId` which determines `id` or `businessKey`.

So `UnitType` API model `retrieve` invokes AbstractJpaIdService class, so I modified the underlying `MapperUtil.getEntityByApiId` method. But instead I could change it's `getEntity` method to invoke `MapperUtil.getEntityByKeyId`. 

You agree with this?